src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -143,14 +145,31 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
                 xRayTraceEvent.subsegments[0].http.response = {
                     status: response.status
                 };
+
+                if (is429(response.status)) {
+                    xRayTraceEvent.subsegments[0].throttle = true;
```

Comment:
question: I'm assuming `subsegments[0]` refers to the very first downstream call from the application. If that call returns an error/fault/throttle, we update `subsegments[0]` accordingly. How is `subsegment[1..n]` updated?

Reply:
Looking at the code, it seems like we're responsible for generating the subsegment of the first downstream call, which makes sense.

Reply:
Correct -- there is only one subsegment.